tests/test_runner/test_checkpoint.py

```diff
@@ -221,6 +221,77 @@ def __init__(self):
     os.remove(checkpoint_path)
 
 
+def test_load_checkpoint_metadata():
+    import os
+    import tempfile
+
+    from mmcv.runner import load_checkpoint, save_checkpoint
+
+    class ModelV1(nn.Module):
+
+        def __init__(self):
+            super().__init__()
+            self.block = Block()
+            self.conv1 = nn.Conv2d(3, 3, 1)
+            self.conv2 = nn.Conv2d(3, 3, 1)
+            nn.init.normal_(self.conv1.weight)
+            nn.init.normal_(self.conv2.weight)
+
+    class ModelV2(nn.Module):
+        _version = 2
+
+        def __init__(self):
+            super().__init__()
+            self.block = Block()
+            self.conv0 = nn.Conv2d(3, 3, 1)
+            self.conv1 = nn.Conv2d(3, 3, 1)
+            nn.init.normal_(self.conv0.weight)
+            nn.init.normal_(self.conv1.weight)
+
+        def _load_from_state_dict(self, state_dict, prefix, local_metadata,
+                                  *args, **kwargs):
+            """load checkpoints."""
+
+            # Names of some parameters in has been changed.
+            version = local_metadata.get('version', None)
+            if version is None or version < 2:
+                state_dict_keys = list(state_dict.keys())
```

Comment:
```python
                state_dict_keys = list(state_dict.keys())
                for k in state_dict_keys:
                    convert_map = {'conv1': 'conv0', 'conv2': 'conv1'}
                    for ori_pattern, convert_pattern in convert_map.items():
                        if k.startswith(prefix + ori_pattern):
                            convert_key = k.replace(ori_pattern,
                                                    convert_pattern)
                            state_dict[convert_key] = state_dict[k]
                            del state_dict[k]
```

-> 

```python
                convert_map = {'conv1': 'conv0', 'conv2': 'conv1'}
                for k in state_dict.keys():
                    for ori_str, new_str in convert_map.items():
                        if k.startswith(prefix + ori_str):
                            new_key = k.replace(ori_str, new_str)
                            state_dict[new_key] = state_dict[k]
                            del state_dict[k]
```

Reply:
I think we cannot use `state_dict.keys()` in iteration.
For example:
```python
>>> d = dict(a=1, b=2, c=3)
>>> for k in d.keys():
...     print(k)
...     del d[k]
a
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-555253fa01ee> in <module>
----> 1 for k in d.keys():
      2     print(k)
      3     del d[k]

RuntimeError: dictionary changed size during iteration
```

Reply:
The variable names change requests have been done.

Reply:
got it